kernel/src/engine/arrow_utils.rs

```diff
@@ -1066,6 +1074,27 @@ mod tests {
         assert_eq!(result.column(2).null_count(), 2);
     }
 
+    #[test]
+    fn test_parse_json_with_long_strings() -> DeltaResult<()> {
+        // See issue#1139: https://github.com/delta-io/delta-kernel-rs/issues/1139
+        let schema = Arc::new(ArrowSchema::new(vec![ArrowField::new(
+            "long_val",
+            ArrowDataType::Utf8,
+            true,
+        )]));
+
+        let long_string = "a".repeat(1_000_000); // 1MB string
+        let json_string = format!(r#"{{"long_val": "{}"}}"#, long_string);
+        let input: Vec<Option<&str>> = vec![Some(&json_string)];
+
+        let batch = parse_json_impl(&input.into(), schema.clone())?;
+
+        assert_eq!(batch.num_rows(), 1);
```

Comment:
At least one of these blank lines could probably go away?
```suggestion
        let input: Vec<Option<&str>> = vec![Some(&json_string)];
        let batch = parse_json_impl(&input.into(), schema.clone())?;
        assert_eq!(batch.num_rows(), 1);
```